### PR TITLE
feat(sdk): regex.match

### PR DESCRIPTION
**Pull request by harsh9607**
An implementation to have regex introduced into Winglang
Motivation 
[first step to Add Regex type to String.replace() #3644 ]
Solution 
I have implemented regex match function by using the match in Javascript. 
It takes in the userString and regexPattern string (example below) and returns a true if they match ; false otherwise. 
 
```
let userString   = "peach";
let regexPattern =  "p[a-z]+h";
let matches      = userString.match(regexPattern); 
assert(matches == true);
```
It returns a boolean value of true or false.

 ## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/regex/regex.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Regex utilities and functions
+ *
+ */
+export class Util {
+  /**
+   * regex pattern matching 
+   * @param pattern - regex pattern 
+   * @param userString - given input string
```
Comment (Chriscbr) [inline libs/wingsdk/src/regex/regex.ts:11]:
```suggestion
   * @param text - given input string 
```

File: libs/wingc/src/lib.rs [hunk 2]
```diff
@@ -71,14 +71,16 @@ const WINGSDK_HTTP_MODULE: &'static str = "http";
...
 
-const WINGSDK_BRINGABLE_MODULES: [&'static str; 6] = [
+const WINGSDK_BRINGABLE_MODULES: [&'static str; 7] = [
 	WINGSDK_CLOUD_MODULE,
 	WINGSDK_UTIL_MODULE,
 	WINGSDK_HTTP_MODULE,
 	WINGSDK_MATH_MODULE,
 	WINGSDK_AWS_MODULE,
 	WINGSDK_EX_MODULE,
+    WINGSDK_REGEX_MODULE
```
Comment (Chriscbr) [inline libs/wingc/src/lib.rs:83]: In Rust the code needs to be indented with tabs (I think this is causing the build error)
```suggestion
	WINGSDK_REGEX_MODULE
```

**Review by Chriscbr — CHANGES_REQUESTED** [review]
Thanks @harsh9607, this looks great! I've requested a few changes - please take a look when you have a chance and once they are addressed we will be able to ship the pull request 🎉

File: examples/tests/sdk_tests/regex/match.w [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+assert(matches9 == false);
+assert(matches10 == false);
+
+test "inflight match" {
+        
+let matches1 = regex.match("p[a-z]+ch" , "peach");
+let matches2 = regex.match("[0-9]+" , "0923");
+let matches3 = regex.match("[0-9]+" , "0a923");
+let matches4 = regex.match("^([a-zA-Z0-9_.-]+)@[a-z]+.[a-z]+\$" , "james_bond007@wing.com");
```
[the remark this answers is not in the document]
Comment (harsh9607) [reply examples/tests/sdk_tests/regex/match.w]: Done !

File: libs/wingsdk/src/regex/regex.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { InflightClient } from "../core";
+
+/**
+ * Regex utilities and functions
+ *
```
[the remark this answers is not in the document]
Comment (harsh9607) [reply libs/wingsdk/src/regex/regex.ts:5]: Done !

File: libs/wingsdk/src/regex/regex.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export class Util {
+  /**
+   * regex pattern matching
```
[the remark this answers is not in the document]
Comment (harsh9607) [reply libs/wingsdk/src/regex/regex.ts:9]: Done !

File: libs/wingsdk/src/regex/regex.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+export class Util {
+  /**
+   * regex pattern matching 
+   * @param pattern - regex pattern 
+   * @param userString - given input string 
+   * @returns boolean value true or false;
+   */
+  public static match(pattern: string , userString: string ): boolean {
+    const regex = new RegExp(pattern);
+    if (userString.match(regex) == null ) {
```
[the remark this answers is not in the document]
Comment (harsh9607) [reply libs/wingsdk/src/regex/regex.ts:16]: Done !

File: libs/wingsdk/src/regex/regex.ts (hunk 1 above)
Comment (harsh9607) [reply libs/wingsdk/src/regex/regex.ts]: Done !

File: libs/wingc/src/lib.rs (hunk 2 above)
Comment (harsh9607) [reply libs/wingc/src/lib.rs]: All 5 changes now complete

File: libs/wingsdk/src/regex/regex.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Regex utilities and functions
+ *
+ */
+export class Util {
+  /**
+   * regex pattern matching 
+   * @param pattern - regex pattern 
+   * @param userString - given input string 
+   * @returns boolean value true or false;
```
[the remark this answers is not in the document]
Comment (harsh9607) [reply libs/wingsdk/src/regex/regex.ts]: done.

File: libs/wingc/src/lib.rs [hunk 8]
```diff
@@ -71,14 +71,16 @@ const WINGSDK_HTTP_MODULE: &'static str = "http";
...
 
-const WINGSDK_BRINGABLE_MODULES: [&'static str; 6] = [
+const WINGSDK_BRINGABLE_MODULES: [&'static str; 7] = [
 	WINGSDK_CLOUD_MODULE,
 	WINGSDK_UTIL_MODULE,
 	WINGSDK_HTTP_MODULE,
 	WINGSDK_MATH_MODULE,
 	WINGSDK_AWS_MODULE,
 	WINGSDK_EX_MODULE,
+    	WINGSDK_REGEX_MODULE,
```
Comment (Chriscbr) [inline libs/wingc/src/lib.rs:83]:
```suggestion
	WINGSDK_REGEX_MODULE,
```

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**